### PR TITLE
Dashboard Cards: Display latest pages in Pages card

### LINKED ISSUE #20440: Dashboard Cards: Display 3 latest pages in Pages card
## Description

Data can either be from the dashboard cards endpoint or from the pages endpoint. This is still to be determined.

### PR DESCRIPTION
**Pull request by hassaanelgarem**
Closes #20440
Design ref: YSMw2nbFAgPpjZLLbYvTfT-fi-96_3834

## Description
Displays the 3 most recently modified pages in the Pages card.

## Known Issues
- There is an issue where the date modified is not set correctly when scheduling a page from the app. This can affect the order of the pages, causing the scheduling page to always appear at the top. This will be handled as part of #20521 
- Dark mode is not supported yet. This will be handled as part of #20523
- Sites with no pages are not properly supported yet. This will be handled in #20441

## Testing Instructions

1. Run the Jetpack app
2. Enable the Pages Card Remote Feature Flag from the debug menu
3. Navigate to the dashboard
4. Make sure that the Pages card displays your site's pages
5. Make sure that all page types are displayed (Published, Scheduled, Draft)
6. Make sure that the pages are ordered by the last modified date.
7. Make sure that the page cell matches the design
8. Tap on the floating button
9. Tap Site Page 
10. Fill in the title and content
11. Publish the page
12. Make sure the new page is added to the Pages card

## Regression Notes
1. Potential unintended areas of impact
Dashboard posts card and posts syncing.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing & `DashboardPostsSyncManagerTests`

3. What automated tests I added (or what prevented me from doing so)
`DashboardPostsSyncManagerTests.testSuccessfulPagesSync`

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Pages/DashboardPageCell.swift [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+        // Icon Image View Constraints
+        iconImageView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: Metrics.iconLeadingSpace).isActive = true
+        iconImageView.centerYAnchor.constraint(equalTo: centerYAnchor).isActive = true
+
+        // Title label Constraints
+        titleLabel.leadingAnchor.constraint(equalTo: iconImageView.trailingAnchor, constant: Metrics.titleLabelMargins.leading).isActive = true
+        titleLabel.topAnchor.constraint(equalTo: topAnchor, constant: Metrics.titleLabelMargins.top).isActive = true
+        titleLabel.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -Metrics.titleLabelMargins.bottom).isActive = true
+        titleLabel.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -Metrics.titleLabelMargins.trailing).isActive = true
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Pages/DashboardPageCell.swift]: [nit] WDYT of using NSConstraintLayout.activate instead?

From [Apple docs](https://developer.apple.com/documentation/uikit/nslayoutconstraint/1526955-activate)
> This convenience method provides an easy way to activate a set of constraints with one call. The effect of this method is the same as setting the [isActive](https://developer.apple.com/documentation/uikit/nslayoutconstraint/1527000-isactive) property of each constraint to true. **Typically, using this method is more efficient than activating each constraint individually.**

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/DashboardPostsListCardCell.swift [hunk 2]
```diff
@@ -19,7 +19,7 @@ class DashboardPostsListCardCell: UICollectionViewCell, Reusable {
     private let frameView = BlogDashboardCardFrameView()
 
     lazy var tableView: UITableView = {
-        let tableView = PostCardTableView()
+        let tableView = DashboardCardTableView()
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/DashboardPostsListCardCell.swift:22]: Nice! ✨

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Helpers/DashboardPostsSyncManager.swift [hunk 3]
```diff
@@ -1,11 +1,20 @@
 import Foundation
 
 protocol DashboardPostsSyncManagerListener: AnyObject {
-    func postsSynced(success: Bool, blog: Blog, posts: [AbstractPost]?, for statuses: [String])
+    func postsSynced(success: Bool,
+                     blog: Blog,
+                     postType: DashboardPostsSyncManager.PostType,
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Helpers/DashboardPostsSyncManager.swift:6]: Glad you were able to reuse this for pages 👍

**Review by momo-ozawa — COMMENTED** [review]
Works as described - but I noticed that for sites that have 0 pages, the tableview height isn't adjusted on the initial load. Do you know what might be causing this? 




I reviewed the code by comparing it against the posts code. Since the pages logic is pretty complex, I wonder if we should start assigning QEs to review PRs? I remember doing that for the posts dashboard card project.

**Comment by hassaanelgarem** [comment]
> I noticed that for sites that have 0 pages, the tableview height isn't adjusted on the initial load

@momo-ozawa Yes. I had this listed as a known issue, and I plan to tackle it in #20441 

> Since the pages logic is pretty complex, I wonder if we should start assigning QEs to review PRs?

Good point! @startuptester, I added you as a reviewer, let me know if it makes sense for you to test this PR that only displays the pages. Or do you prefer to wait for the [handling tap actions](https://github.com/wordpress-mobile/WordPress-iOS/issues/20442) PR?

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Pages/DashboardPageCell.swift (hunk 1 above)
Comment (hassaanelgarem) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Pages/DashboardPageCell.swift]: Good point! Updated in 9b0ec64d71942a5c64862b9c5325ad206393d07c 👍

**Comment by momo-ozawa** [comment]
> @momo-ozawa Yes. I had this listed as a known issue, and I plan to tackle it in https://github.com/wordpress-mobile/WordPress-iOS/issues/20441

Ah I missed that - sounds good!

**Review by momo-ozawa — APPROVED** [review]

**Comment by startuptester** [comment]
Thank you @momo-ozawa  and @hassaanelgarem!  
I added tests from our list in the Project Spec and this time i ran through them to familiarize myself with the card, but normally I'd expect these to be done as part of the dev and PR review for a UI change. 😀 🙌

- [ ] pages with long title - technically it adapts but it's almost illegible. IMO, truncating would be more effective? 
IMG_2223
- [x] pages with no title - `no title` 
- [x] special characters 
- [x] RTL (just seeing how it looks flipped at this point. I checked Hebrew)
- [x] screen rotation 
- [x] new pages added 
- [ ] context menu options - I couldn't find a way to bring it back after hiding? Maybe this part isn't done yet?
- [x] I moved pages through various tag states 
- [x] VoiceOver
- [x] Accessibility Fonts 
- [x] High Contrast
- [ ] Bold Text
This only breaks the first time you choose that site. Subsequent times it works as expected. 
IMG_2226

**Comment by hassaanelgarem** [comment]
> normally I'd expect these to be done as part of the dev and PR review for a UI change

@startuptester Thanks for the review, and we'll keep that in mind 👍 

> pages with long title - technically it adapts but it's almost illegible

Good point. Fixed in f806be2a0ce2f08029ca2ddce3edc3d6080f7c58 👍 

> context menu options - I couldn't find a way to bring it back after hiding? Maybe this part isn't done yet?

It's done, but it depends on another feature that's not enabled yet. You'll find testing instructions in #20495. Tl;dr, Enable the Personalize Home Tab Feature Flag from the debug menu, and you should see a card at the bottom of the dashboard from which you can bring back hidden cards.  

> Bold Text - This only breaks the first time you choose that site. Subsequent times it works as expected.

I'm unable to reproduce this. But it seems related to the [known issue with sites with no pages](https://github.com/wordpress-mobile/WordPress-iOS/pull/20524#pullrequestreview-1383171571). As mentioned, I plan to tackle this in a different PR. Does this happen for you only with Bold Text? And how consistent is it?

**Comment by startuptester** [comment]
I think you're right, @hassaanelgarem - that does look similar! I was only able to repro it on a fresh install, and refreshing cleared the issue. 
https://user-images.githubusercontent.com/66104034/231946495-1568c8dc-2c51-4c63-9cab-b167b1b5b5f0.MP4

**Pull request merged**